File: C++/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> a(nums1.begin(), nums1.end());
        set<int> b;
        for (int n : nums2) {
            if (a.count(n)) {
                b.insert(n);
            }
        }
        vector<int> ans(b.begin(), b.end());
        return ans;
    }
};

File: Go/intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {
    m := make(map[int]byte)
    for _, n := range nums1 {
        m[n] = 1
    }
    for _, n := range nums2 {
        if m[n] == 1 {
            m[n] = 2
        }
    }
    result := make([]int, 0, len(m))
    for k, v := range m {
        if v == 2 {
            result = append(result, k)
        }
    }
    return result
}